pkg/e2e/podplacement/pod_placement_test.go

```diff
@@ -174,13 +174,135 @@ var _ = Describe("The Pod Placement Operand", func() {
 				g.Expect(pod.Annotations).NotTo(BeEmpty())
 				g.Expect(pod.Annotations).To(HaveKeyWithValue("openshift.io/scc", "privileged"))
 			})
-			verifyPodNodeAffinity(ns, "app", "test", utils.ArchitectureAmd64,
+			verifyPodNodeAffinity(ns, "app", "test", utils.ArchLabel, corev1.NodeSelectorOpIn, utils.ArchitectureAmd64,
 				utils.ArchitectureArm64, utils.ArchitectureS390x, utils.ArchitecturePpc64le)
 		})
+		It("should set the node affinity when users node affinity non conflict", func() {
+			var err error
+			ns := framework.NewEphemeralNamespace()
+			err = client.Create(ctx, ns)
+			Expect(err).NotTo(HaveOccurred())
+			//nolint:errcheck
+			defer client.Delete(ctx, ns)
+			d := appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-deployment",
+					Namespace: ns.Name,
+				},
+				Spec: appsv1.DeploymentSpec{
+					Replicas: utils.NewPtr(int32(1)),
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"app": "test",
+						},
+					},
+					Template: corev1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								"app": "test",
+							},
+						},
+						Spec: corev1.PodSpec{
+							Affinity: &corev1.Affinity{
+								NodeAffinity: &corev1.NodeAffinity{
+									RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+										NodeSelectorTerms: []corev1.NodeSelectorTerm{
+											{
+												MatchExpressions: []corev1.NodeSelectorRequirement{
+													{
+														Key:      utils.OSLabel,
+														Operator: corev1.NodeSelectorOpIn,
+														Values: []string{
+															utils.OSLinux,
+														},
+													},
+												},
+											},
```

Comment:
```suggestion
											{
												MatchExpressions: []corev1.NodeSelectorRequirement{
													{
														Key:      utils.OSLabel,
														Operator: corev1.NodeSelectorOpIn,
														Values: []string{
															utils.OSLinux,
														},
													},
												},
											},
											{
												{
														Key:      utils.ArchLabel,
														Operator: corev1.NodeSelectorOpIn,
														Values: []string{
															utils.ArchitectureS390X,
														},
													},
												},
											},
											
											
```

Regarding https://github.com/openshift/multiarch-manager-operator/pull/44/files#r1498687311, in this case, we expect that the first NodeSelectorTerm gets the automated arch-aware affinity.

The second (already including an arch-specific matchExpression) should not as we don't want to override any spec provided by users.

Reply:
yes this is what i meant. sorry if I wasn't clear